romancal/outlier_detection/outlier_detection.py

```diff
@@ -84,22 +84,27 @@ def do_detection(self):
             # Start by creating resampled/mosaic images for
             # each group of exposures
             resamp = resample.ResampleData(
-                self.input_models, single=True, blendheaders=False, **pars
+                self.input_models, single=False, blendheaders=False, **pars
             )
             drizzled_models = resamp.do_drizzle()
 
         else:
             # for non-dithered data, the resampled image is just the original image
-            drizzled_models = self.input_models
-            for i in range(len(self.input_models)):
-                drizzled_models[i].weight = build_driz_weight(
+            drizzled_models = ModelContainer()
+            for i, model in enumerate(self.input_models):
+                model["weight"] = build_driz_weight(
                     self.input_models[i],
                     weight_type="ivm",
                     good_bits=pars["good_bits"],
                 )
+                drizzled_models.append(model)
```

Comment:
Just because I'm not seeing it... can you elaborate on the choice of directly iterating on self.input_models vs. making a new model container for them?

Reply:
Thanks, Eddie!
When I used the `ModelContainer`, I intended to test something else not related directly with the ticket; it should not be there. What I really wanted to do is to clean up the `for` block to look like this:
``` python
drizzled_models = self.input_models
for model in drizzled_models:
    model.weight = build_driz_weight(
        model,
        weight_type="ivm",
        good_bits=pars["good_bits"],
     )
```